awx/ui_next/src/screens/NotificationTemplate/NotificationTemplateDetail/NotificationTemplateDetail.jsx

```diff
@@ -358,4 +371,164 @@ function NotificationTemplateDetail({ i18n, template }) {
   );
 }
 
+function CustomMessageDetails({ messages, defaults, type, i18n }) {
+  const showMessages = type !== 'webhook';
+  const showBodies = ['email', 'pagerduty', 'webhook'].includes(type);
+
+  return (
+    <>
+      {showMessages && (
+        <CodeDetail
+          label={i18n._(t`Start message`)}
+          value={messages.started.message || defaults.started.message}
+          mode="jinja2"
+          rows="2"
+          fullWidth
+        />
+      )}
+      {showBodies && (
+        <CodeDetail
+          label={i18n._(t`Start message body`)}
+          value={messages.started.body || defaults.started.body}
+          mode="jinja2"
+          rows="6"
+          fullWidth
+        />
+      )}
+      {showMessages && (
+        <CodeDetail
+          label={i18n._(t`Success message`)}
+          value={messages.success.message || defaults.success.message}
+          mode="jinja2"
+          rows="2"
+          fullWidth
+        />
+      )}
+      {showBodies && (
+        <CodeDetail
+          label={i18n._(t`Success message body`)}
+          value={messages.success.body || defaults.success.body}
+          mode="jinja2"
+          rows="6"
+          fullWidth
+        />
+      )}
+      {showMessages && (
+        <CodeDetail
+          label={i18n._(t`Error message`)}
+          value={messages.error.message || defaults.error.message}
+          mode="jinja2"
+          rows="2"
+          fullWidth
+        />
+      )}
+      {showBodies && (
+        <CodeDetail
+          label={i18n._(t`Error message body`)}
+          value={messages.error.body || defaults.error.body}
+          mode="jinja2"
+          rows="6"
+          fullWidth
+        />
+      )}
+      {showMessages && (
+        <CodeDetail
+          label={i18n._(t`Workflow approved message`)}
+          value={
+            messages.workflow_approval.approved.message ||
```

Comment:
```suggestion
            messages?.workflow_approval?.approved?.message ||
```

@keithjgrant 
`messages.workflow_approval` can be `null` so I think we need to check before referencing its properties or use optional chaining in the various places it is referenced:

![Screenshot from 2020-09-14 13-17-15](https://user-images.githubusercontent.com/9753817/93117496-f5ec9b80-f68c-11ea-94bc-3ef1e5739673.png)

![Screenshot from 2020-09-14 13-18-06](https://user-images.githubusercontent.com/9753817/93117563-0866d500-f68d-11ea-9ba2-55a556fabc66.png)

Reply:
probably needed here as well: https://github.com/ansible/awx/blob/devel/awx/ui_next/src/screens/NotificationTemplate/shared/NotificationTemplateForm.jsx#L144-L170